pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -113,10 +144,27 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(proxy *model.Proxy, push *mo
 	}
 	resources = cb.normalizeClusters(resources)
 
-	if cacheStats.empty() {
+	if clusterCacheStats.empty() {
 		return resources, model.DefaultXdsLogDetails
 	}
-	return resources, model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", cacheStats.hits, cacheStats.hits+cacheStats.miss)}
+	return resources, model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", clusterCacheStats.hits, clusterCacheStats.hits+clusterCacheStats.miss)}
+}
+
+func allConfigKeysOfType(keys []model.ConfigKey, cfg config.GroupVersionKind) bool {
```

Comment:
```suggestion
func allConfigKeysAreSameType(keys []model.ConfigKey, cfg config.GroupVersionKind) bool {
```

currently sounds like its filtering the config keys to only the `cfg` type